### PR TITLE
Move ReportTable component to separate file.

### PR DESCRIPTION
**Pull request by MarkAZhang**
Follow-up PR.

### CODE REVIEW

**Comment by jshoe** [comment]
not sure about best practices but should we go with the index.jsx naming? @tfrcarvalho

File: app/assets/src/components/views/report/ReportTable/DetailCells.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/report/ReportTable/DetailCells.jsx]: Current convention is to keep folders in snake case, so: `ReportTable` -> `report_table`

File: app/assets/src/components/views/report/ReportTable/index.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import Tipsy from "react-tipsy";
+
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/report/ReportTable/index.jsx]: Remove blank lines between imports.

Is this order following  some logic of imports? Like third-party imports, idseq components and same package imports?

File: app/assets/src/components/views/report/ReportTable/DetailCells.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+function DetailCells({ parent, openTaxonModal }) {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/report/ReportTable/DetailCells.jsx]: Our convention is to use the format: `const DetailCells = ({parent, openTaxonModal}) => {...}`

**Comment by MarkAZhang** [comment]
I looked at what the Semantic UI React and Material UI React Github repos do.

The best practice seems to be to have `index.jsx` be a simple redirect file:
`export default from "./ReportTable"`

And have the meat of the code be in `ReportTable.jsx`.

This allows other components to import ReportTable with:
`import ReportTable from "./components/ReportTable"`

instead of:
`import ReportTable from "./components/ReportTable/Reportable"`.

I'll make this change in my file. Thanks for checking!

File: app/assets/src/components/views/report/ReportTable/index.jsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                          ? "active column-switcher"
+                          : "column-switcher"
+                      }
+                      onClick={() => {
+                        parent.setState({ countType: "NT" });
+                      }}
+                    >
+                      NT
+                    </div>
+                    <div
```
[the remark this answers is not in the document]
Comment (MarkAZhang) [reply app/assets/src/components/views/report/ReportTable/index.jsx]: Next commit.

File: app/assets/src/components/views/report/ReportTable/index.jsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+              {parent.render_column_header(
+                "%conc",
+                `${parent.state.countType}_percentconcordant`,
+                `Percentage of aligned reads belonging to a concordantly mappped pair (NCBI NT/NR)`,
+                parent.showConcordance
+              )}
+              <th>
+                <Tipsy content="Switch count type" placement="top">
+                  <div className="sort-controls center left">
+                    <div
```
[the remark this answers is not in the document]
Comment (MarkAZhang) [reply app/assets/src/components/views/report/ReportTable/index.jsx]: Next commit.

File: app/assets/src/components/views/report/ReportTable/index.jsx (hunk 2 above)
Comment (MarkAZhang) [reply app/assets/src/components/views/report/ReportTable/index.jsx]: Discussed in person to keep blank lines.

**Pull request merged**